backend/dataall/core/permissions/db/permission_repositories.py

```diff
@@ -73,48 +75,54 @@ def save_permission(session, name: str, description: str, permission_type: str)
         return permission
 
     @staticmethod
-    def init_permissions(session):
-        perms = []
-        count_resource_permissions = (
-            session.query(models.Permission).filter(models.Permission.type == PermissionType.RESOURCE.name).count()
-        )
-
-        logger.debug(
-            f'count_resource_permissions: {count_resource_permissions}, RESOURCES_ALL: {len(permissions.RESOURCES_ALL_WITH_DESC)}'
+    def init_permissions(session: Session) -> [str]:
+        return Permission._check_and_save_permissions(
+            session,
+            [
+                perm.name
+                for perm in session.query(models.Permission)
+                .filter(models.Permission.type == PermissionType.RESOURCE.name)
+                .all()
+            ],
+            permissions.RESOURCES_ALL_WITH_DESC,
+            PermissionType.RESOURCE,
+        ) + Permission._check_and_save_permissions(
+            session,
+            [
+                perm.name
+                for perm in session.query(models.Permission)
+                .filter(models.Permission.type == PermissionType.TENANT.name)
+                .all()
+            ],
+            permissions.TENANT_ALL_WITH_DESC,
+            PermissionType.TENANT,
         )
 
-        if count_resource_permissions < len(permissions.RESOURCES_ALL_WITH_DESC):
-            for name, desc in permissions.RESOURCES_ALL_WITH_DESC.items():
-                perms.append(
-                    Permission.save_permission(
-                        session,
-                        name=name,
-                        description=desc,
-                        permission_type=PermissionType.RESOURCE.name,
-                    )
-                )
-                logger.info(f'Saved permission {name} successfully')
-            logger.info(f'Saved {len(perms)} resource permissions successfully')
+    @staticmethod
+    def _check_and_save_permissions(
+        session: Session, db_perms: [str], app_perms: dict[str, str], perm_type: PermissionType
+    ):
+        perms = []
+        logger.info(f'db  perms {sorted(db_perms)}')
+        logger.info(f'app perms {sorted(app_perms)}')
 
-        count_tenant_permissions = (
-            session.query(models.Permission).filter(models.Permission.type == PermissionType.TENANT.name).count()
-        )
+        for perm, freq in Counter(db_perms).items():
+            if freq > 1:
+                logger.error('%50s permission appears %3d times in the db', perm, freq)
 
```

Comment:
I would also avoid deleting permissions from RDS because we do not know if those permissions are used down the line, is this what you were thinking when you implemented the warnings?

Reply:
I thought about deleting directly but we would have to make sure no URI is used somewhere else and if it does we need to update it to the one we are planning to keep. One thought is to write a DB migration script that will always execute and clean up the dips. Another one is to write one that converts all the URIs to be equal with the permission name, that way we both deduplicate the existing store but also future proofing it (uri is the primary key so we won't be able to insert dups)